process/block/shardblock.go

```diff
@@ -545,7 +552,14 @@ func (sp *shardProcessor) indexBlockIfNeeded(
 			continue
 		}
 		signersIndexes = sp.nodesCoordinator.GetValidatorsIndexes(publicKeys)
-		go sp.core.Indexer().SaveRoundInfo(int64(i), shardId, signersIndexes, true)
+		roundInfo = indexer.RoundInfo{
+			SignersIndexes:   signersIndexes,
+			BlockWasProposed: true,
+			ShardId:          shardId,
+			Timestamp:        time.Duration(header.GetTimeStamp() - (currentBlockRound - i)),
```

Comment:
I would add the correct round time by computing the round duration and round time from current and previous block  header. These should normally come from rounder, but seems it is not available to block processor.

Maybe add a TODO to inject the rounder interface to block processor

Reply:
Done.